### ISSUE TITLE
Lint on use of `.into()` from `T` to `Option<T>`

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust-clippy#11950]
### What it does

Flags all usage of `.into()` to wrap a value in `Some` and turn it into an `Option`.

### Advantage

I've found myself wanting to avoid the use of `.into()` to convert from `T` to `Option<T>`. It feels like it decreases readability, because it's a blanket impl, and `.into()` could just as easily be converting into a completely different type that's specific to `T`. Given that, I'd like to avoid such usage in favor of writing `Some` explicitly instead. I'd love to have a lint that helps me do that.

I'd expect this to be a style or restriction lint, allow-by-default, since this is a matter of taste and style rather than something universal that everyone would want.

### Drawbacks

This is a lint enforcing developer style, which would only be used if explicitly allowed, so I wouldn't expect any drawbacks for those using it; it should produce what developers expect.

### Example

```rust
struct S(Option<u64>);
let x = 42;
let y = S(x.into());
```

Could be written as:

```rust
struct S(Option<u64>);
let x = 42;
let y = S(Some(x));
```
